web/src/pages/Dashboard.tsx

```diff
@@ -175,14 +175,21 @@ const StatsWrapper = () => {
     useRecoilState(selectedCity);
   const [totalSelectableCities, setTotalSelectableCities] =
     React.useState(selectedCities);
+  const { updateCityList } = useUpdateCityList();
 
   React.useEffect(() => {
-    const selectableCityIds = selectableCitiesData.map((c) => c.id);
+    // This function will only be called once, when the component is first rendered
+    updateCityList();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we should avoid using eslint-disabled and rather investigate and fix the dependencies issue. I made the investigation and you have here a working version, please take a look (and cleanup if you're using it): 
https://github.com/singlestore-labs/demo-realtime-digital-marketing/commit/db07e0bd11989a3240baed9dc92aa454a69f2e8b

Reply:
The issue here is If you have any values in an array of use affect the function will be called every time it rerenders the component.

If empty, it will be called only on the first mount. Another way is to use a variable that will say if its first mount or data was loaded once

I have just pushed the variable way let me know if that is okay to use.

Reply:
@sandeshvijayraj did you see the version I sent to you? What do you think?

I don't think we need an extra variable here, but rather to fix the dependencies that are causing the re render. Is there any reason to only update the city list once on the update?

Reply:
That goes to infinite loop because the the function ref changes on every rerender and useEffect considered functions ref as a change in function causing an infinite loop

BTW I avoided a variable by memorizing the function ref itself to avoid changing the ref and causing an infinite loop

This should be good now

Reply:
I think just got better and clean way to handle

Now using selector in recoil that will automatically set default value by calling the getCities API

Reply:
I like the new approach!